internal/testutil/datagen/datagen.sh

```diff
@@ -0,0 +1,250 @@
+# Values can be overridden using this and --set commands, or -f and point to a custom values file.
+# Useful to tweak things specific to minikube, or toggle specific features.
+# See  ../../../e2e/charts/e2e-resources/values.yaml for more details.
+HELM_E2E_ARGS=""
+
+# scrapper_selector is the label with which the scraper deployment is deployed.
+scrapper_selector="app=scraper"
+scrapper_namespace="mock"
+
+# main subcommand runs the whole flow of the script: Bootstrap, scrape, and cleanup
+function main() {
+    if [[ -z "$1" ]]; then
+        echo "Usage: $0 <output_folder>"
+        exit 1
+    fi
+
+    # Install scraper pod and e2e resources.
+    bootstrap
+
+    # Scraper pod is a deployment, so we need to locate it.
+    pod=$(scraper_pod "$scrapper_selector")
+    if [[ -z "pod" ]]; then
+        echo "Could not find scraper pod (-l $scrapper_selector)"
+        exit 1
+    fi
+
+    # Dump test info
+    mkdir -p "$1" && cd "$1"
+    testinfo > README.md
+
+    # Kubelet endpoints
+    mkdir -p kubelet
+    echo "Extracting kubelet /pods"
+    kubectl exec -n $scrapper_namespace $pod -- datagen.sh scrape kubelet pods > kubelet/pods
+
+    echo "Extracting kubelet /metrics/cadvisor"
+    mkdir -p kubelet/metrics
+    kubectl exec -n $scrapper_namespace $pod -- datagen.sh scrape kubelet metrics/cadvisor > kubelet/metrics/cadvisor
+
+    echo "Extracting kubelet /stats/summary"
+    mkdir -p kubelet/stats
+    kubectl exec -n $scrapper_namespace $pod -- datagen.sh scrape kubelet stats/summary > kubelet/stats/summary
+
+    # KSM endpoint
+    mkdir -p ksm
+    echo "Extracting ksm /metrics"
+    kubectl exec -n $scrapper_namespace $pod -- datagen.sh scrape ksm > ksm/metrics
+
+    if [[ "$DISABLE_CONTROLPLANE" != "1" ]]; then
+        # Control plane components
+        mkdir -p controlplane/api-server
+        echo "Extracting api-server /metrics"
+        kubectl exec -n $scrapper_namespace $pod -- datagen.sh scrape controlplane apiserver > controlplane/api-server/metrics
+
+        mkdir -p controlplane/etcd
+        echo "Extracting etcd /metrics"
+        kubectl exec -n $scrapper_namespace $pod -- datagen.sh scrape controlplane etcd > controlplane/etcd/metrics
+
+        mkdir -p controlplane/controller-manager
+        echo "Extracting controller-manager /metrics"
+        kubectl exec -n $scrapper_namespace $pod -- datagen.sh scrape controlplane controllermanager > controlplane/controller-manager/metrics
+
+        mkdir -p controlplane/scheduler
+        echo "Extracting scheduler /metrics"
+        kubectl exec -n $scrapper_namespace $pod -- datagen.sh scrape controlplane scheduler > controlplane/scheduler/metrics
+    else
+        echo "Skipping control plane metrics"
+    fi
+
+    # K8s objects
+    echo "Generating list of kubernetes resources"
+    kubedump nodes
+    kubedump namespaces
+    kubedump endpoints
+    kubedump services
+    kubedump pods
+
+    cd -
```

Comment:
could we save the location and cd to it instead of a generic `cd -`

Reply:
Not sure myself if it is more idiomatic to make `oldwd=$(pwd)` and then `cd $oldwd`, or just `cd -` 🤔 
Summoning our bash expert @kang-makes

Reply:
```bash
#!/bin/bash

function strip_home() {
  sed -e "s|$HOME|\$HOME|" <<<$1
}

echo Before subshell: `strip_home $PWD`
(
  cd $HOME/.local/bin
  echo Inside subshell: `strip_home $PWD`
)
echo After subshell: `strip_home $PWD`
```
So...
```
Before subshell: $HOME/.local/tmp
Inside subshell: $HOME/.local/bin
After subshell: $HOME/.local/tmp
```

Reply:
Wrapped collection in a subshell here: https://github.com/newrelic/nri-kubernetes/pull/269/commits/271a7dd94a2772c7194916cc46bc6e13d6537fc4. WDYT?